compiler/inlineasm.go

```diff
@@ -46,24 +75,12 @@ func (b *builder) createInlineAsm(args []ssa.Value) (llvm.Value, error) {
 func (b *builder) createInlineAsmFull(instr *ssa.CallCommon) (llvm.Value, error) {
 	asmString := constant.StringVal(instr.Args[0].(*ssa.Const).Value)
 	registers := map[string]llvm.Value{}
+
 	if registerMap, ok := instr.Args[1].(*ssa.MakeMap); ok {
-		for _, r := range *registerMap.Referrers() {
-			switch r := r.(type) {
-			case *ssa.DebugRef:
-				// ignore
-			case *ssa.MapUpdate:
-				if r.Block() != registerMap.Block() {
-					return llvm.Value{}, b.makeError(instr.Pos(), "register value map must be created in the same basic block")
-				}
-				key := constant.StringVal(r.Key.(*ssa.Const).Value)
-				registers[key] = b.getValue(r.Value.(*ssa.MakeInterface).X, getPos(instr))
-			case *ssa.Call:
-				if r.Common() == instr {
-					break
```

Comment:
Instead of extracting `collectInlineAsmRegisters`, why not name the outer loop and use `break <name>`? Something like:

```go
referrers:
    for _, r := range *registerMap.Referrers() {
...
        if r.Common() == instr {
	        break referrers
        }
```